newIDE/app/src/EventsFunctionsExtensionEditor/OptionsEditorDialog/UsersAutocomplete.js

```diff
@@ -0,0 +1,149 @@
+//@flow
+import React from 'react';
+import { useState } from 'react';
+import { t } from '@lingui/macro';
+import SemiControlledMultiAutoComplete from '../../UI/SemiControlledMultiAutoComplete';
+import {
+  searchUserPublicProfilesByUsername,
+  type UserPublicProfile,
+  getUserPublicProfilesByIds,
+} from '../../Utils/GDevelopServices/User';
+
+import useForceUpdate from '../../Utils/UseForceUpdate';
+
+type Props = {|
+  eventsFunctionsExtension: gdEventsFunctionsExtension,
+|};
+
+export const UsersAutocomplete = (props: Props) => {
+  const forceUpdate = useForceUpdate();
+  const [authors, setAuthors] = React.useState([]);
+  const [authorInput, setAuthorInput] = useState('');
+  const [loading, setLoading] = useState(false);
+  const [
+    completionUserPublicProfiles,
+    setCompletionUserPublicProfiles,
+  ] = React.useState<Array<UserPublicProfile>>([]);
+  const [authorsError, setAuthorsError] = React.useState(null);
```

Comment:
What I usually do is that I store the `Error` object (or `null` if no error). This allows the UI to render a message specific to the error, or a generic message.

(in this case, type as `React.useState<?Error>(null)`)